R/mod-upload-documentation.R

```diff
@@ -6,110 +6,108 @@
 #'
 #' @noRd
 #' @param id the module id
-#' @param study_link_human html link to example of a study using human data
-#' @param study_link_animal html link to example of a study using animal models
-#' @param study_link_ref html link to example of study acknowledgment
-#'   preference
+#' @param markdown_path path to the markdown file to be displayed
+#' @param include_widget TRUE if the upload widget should be included,
+#'   else FALSE
 #' @return html ui for the module
-upload_documents_ui <- function(id, study_link_human,
-                                study_link_animal, study_link_ref) {
+upload_documents_ui <- function(id, markdown_path, include_widget) {
```

Comment:
```suggestion
upload_documents_ui <- function(id, markdown_path, include_widget = FALSE) {
```
Do you want to default this parameter to FALSE as described in the above `@param` description

Reply:
The `@param` description is explaining what is required to include or exclude the widget, not necessarily that there is a default value.

This is a really great question, though. It made me think about where this is coming from and the importance of it. In short, I want this to throw an error if `include_widget` is not specified (i.e. I don't want a default) because including or excluding it should be a deliberate choice. For the widget to work, the module's server function must be called, as well, but if the widget is not included and the server function is called, then the app will break.

I've updated the `@param` docs to be more descriptive of what's needed depending on the choice. Hopefully this helps clarify a bit.